### PR TITLE
Test Improvements

### PR DESCRIPTION
**Pull request by Ashod**
- wsd: test: better UnitWopiLockReadOnly
- wsd: test: correct usage of string::find
- wsd: test: move File-Serve whitebox tests to own file
- wsd: test: better OnwerTermination test
- wsd: test: new UnitWOPI Overload test
- wsd: test: extend wait on load test 
- wsd: test: exclude readonly sessions from modification indication 
- wsd: test: exit on failure to load unit-test lib 
- wsd: test: fix typo in test name 
- wsd: test: improved single-test runner 
- wsd: test: enable experimental features in tests

### CODE REVIEW

File: scripts/run-wsdunit [hunk 1]
```diff
@@ -13,27 +13,41 @@ if [ -z "$name" ]; then
...
-count=0
-while true
-do
-    count=$((count+1))
-
-    (cd test; ./run_unit.sh --test-name $name.la --log-file $name.log --trs-file $name.trs --color-tests yes --enable-hard-errors yes --expect-failure no -- ./$name.la)
-    ./coolwsd --disable-cool-user-checking --cleanup
-    if ! grep -q FAIL test/$name.trs; then
-        RET=0
-        break
```
Comment (Ashod) [inline scripts/run-wsdunit:27]: This prevented finding out if the test passed sometimes or never.

File: test/UnitWOPI.cpp [hunk 2]
```diff
@@ -139,6 +148,119 @@
...
+                // just wait for the results
+                break;
+            }
+        }
+    }
+};
+
+UnitBase** unit_create_wsd_multi(void)
+{
+    // return new UnitBase* [3] { new UnitWOPI(), new UnitOverload(), nullptr };
```
[the remark this answers is not in the document]
Comment (Ashod) [reply test/UnitWOPI.cpp:262]: Intentionally disabled; this test is for stressing coolwsd, it doesn't have a passing condition.

File: common/Unit.cpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ UnitBase** UnitBase::linkAndCreateUnit(UnitType type, const std::string& unitLib
     DlHandle = dlopen(unitLibPath.c_str(), RTLD_GLOBAL|RTLD_NOW);
     if (!DlHandle)
     {
-        LOG_ERR("Failed to load " << unitLibPath << ": " << dlerror());
+        LOG_ERR("Failed to load unit-test lib " << dlerror());
```
Comment (vmiklos) [inline common/Unit.cpp:63]: Why hiding the unitLibPath?

File: scripts/run-wsdunit (hunk 1 above)
Comment (vmiklos) [reply scripts/run-wsdunit:27]: Heh, it was intentional, but I can probably live with the new behavior as well. Funny, now that I re-read the comment at the top of the file, indeed it describes what you did.

**Review by vmiklos — APPROVED** [review]
Looks OK overall, thanks.

File: common/Unit.cpp (hunk 3 above)
Comment (Ashod) [reply common/Unit.cpp:63]: Turns out `dlerror()` already includes the path, so it was a duplicate.

Comment (vmiklos) [reply common/Unit.cpp:63]: Ah, makes sense. Thanks!

File: test/run_unit.sh.in [hunk 4]
```diff
@@ -112,5 +114,6 @@ else
 	echo "============================================================="
 	echo ":test-result: FAIL $tst" >> $test_output
 fi
+exit 1
```
Comment (vmiklos) [inline test/run_unit.sh.in:117]: Ah, hm, one side-effect of this is that now we stop on the first failure with 'make check', while we used to run all tests and we only reported an error at the very end with all failures. Is this wanted?

I get that it's not nice to grep for an error string in the log, but possibly it's a smaller problem than only seeing till the next test failure now. What do you think?

Comment (Ashod) [reply test/run_unit.sh.in:117]: Agreed. `--fail-fast` was added in 6de46d746e4932b7aa339fb251fc6b5f5cc51413 specifically to address this point.

**Pull request merged**